Mathlib/Lean/Data/RBTree.lean

```diff
@@ -0,0 +1,23 @@
+/-
+Copyright (c) 2023 Joachim Breitner. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joachim Breitner
+-/
+import Lean.Data.RBTree
+
+/-!
+# Additional functions on `Lean.RBTree`.
+-/
+
+universe u
+variable {α : Type u} {cmp : α → α → Ordering}
+
+/-- The intersection of a (non-empty) array of `RBTree`s. If
+the input is empty, the empty tree is returned. -/
+def Lean.RBTree.intersects (ts : Array (RBTree α cmp)) : RBTree α cmp :=
```

Comment:
maybe
```suggestion
def Lean.RBTree.intersectArray (ts : Array (RBTree α cmp)) : RBTree α cmp :=
```
since it's a bit weird to have the name sound like a predicate

Reply:
Good point. The suffix `s` for the n-ary variant of an operation is a Haskell-ism.

Reply:
I think the mathlib style would be `aIntersect` to leave room for `iIntersect` (for a finite indexed family) and `sIntersect` (for a set), but those would be nonsensical here so I don't think you need to follow that pattern.

Of course, if Std is already full of haskell-isms then your name is fine1